File: java/src/main/java/io/dzcode/kuliya/Node.java
package io.dzcode.kuliya;

import org.json.JSONObject;

public class Node extends JSONObject {

  public Node(String nameAr, String nameEn, String nameFr, String univ) {
    JSONObject nameObj = new JSONObject();
    nameObj.put("ar", nameAr);
    nameObj.put("en", nameEn);
    nameObj.put("fr", nameFr);
    this.put("name", nameObj);
    this.put("type", univ);
  }

  @Override
  public boolean equals(Object obj) {

    if (obj == null)
      return false;
    int hash1 = ((JSONObject) obj).toString().hashCode();
    int hash2 = ((JSONObject) this).toString().hashCode();

    System.out.println(((JSONObject) obj).toString());
    System.out.println(((JSONObject) this).toString());

    return hash1 == hash2;
  }

}

File: python/src/kulya_python/node.py
class Node:
    def __init__(self, nameAr: str, nameEn: str, nameFr: str, univ: str):
        self.nameAr = nameAr
        self.nameEn = nameEn
        self.nameFr = nameFr
        self.univ = univ

    def getNameAr(self):
        return self.nameAr

    def getNameEn(self):
        return self.nameEn

    def getNameFr(self):
        return self.nameFr

    def getUniv(self):
        return self.univ
